src/workspace-data/data-table/wds/WDSContent.test.ts

```diff
@@ -0,0 +1,229 @@
+      entityType: 'marble',
+      attributes: { color: 'red' },
+    },
+    {
+      name: '2',
+      entityType: 'marble',
+      attributes: { color: 'yellow' },
+    },
+    {
+      name: '3',
+      entityType: 'marble',
+      attributes: { color: 'green' },
+    },
+  ],
+};
+describe('WDSContent', () => {
+  const setup = ({ props, features, entities }: SetupOptions = defaultSetupOptions) => {
+    const user = userEvent.setup();
+
+    const getPageResponse: DeepPartial<EntityQueryResponse> = {
+      results: entities,
+      resultMetadata: {
+        unfilteredCount: entities.length,
+        filteredCount: entities.length,
+        filteredPageCount: entities.length / 100 + 1,
+      },
+    };
+
+    const dataProvider: DeepPartial<DataTableProvider> = {
+      getPage: jest.fn().mockResolvedValue(getPageResponse),
+      features,
+    };
+
+    return { user, props: { ...props, dataProvider: dataProvider as DataTableProvider } };
+  };
+
+  describe('delete column button', () => {
+    it('is displayed when editable and supportsAttributeDeleting are true', async () => {
+      // Arrange
+      const { user, props } = setup({
+        ...defaultSetupOptions,
+        props: { ...defaultProps, editable: true },
+        features: { ...defaultFeatures, supportsAttributeDeleting: true },
+      });
+
+      await act(() => {
+        render(h(WDSContent, props));
+      });
+
+      const columnMenus = screen.queryAllByRole('button', { name: 'Column menu' });
+      expect(columnMenus.length).toEqual(2);
+      const [_unusedIdColumnMenu, colorColumnMenu] = columnMenus;
+
+      await user.click(colorColumnMenu);
+
+      const deleteColorColumnButton = screen.getByRole('button', { name: 'Delete Column' });
+      const deleteConfirmationButton = screen.queryByTestId('confirm-delete');
+
+      // Assert
+      expect(deleteConfirmationButton).not.toBeInTheDocument();
+      expect(screen.queryByText(/Are you sure you want to delete the column/)).not.toBeInTheDocument();
+
+      // Act
+      await user.click(deleteColorColumnButton);
+
+      // Assert
+      expect(screen.getByRole('dialog')).toHaveTextContent(/Are you sure you want to delete the column/);
+      expect(screen.getByRole('dialog')).toContainElement(screen.getByTestId('confirm-delete'));
+    });
+
+    it('is hidden when editable is true, but supportsAttributeDeleting is false', async () => {
+      // Arrange
+      const { user, props } = setup({
+        ...defaultSetupOptions,
+        props: { ...defaultProps, editable: true },
+        features: { ...defaultFeatures, supportsAttributeDeleting: false },
+      });
+
+      // Act
+      await act(() => {
+        render(h(WDSContent, props));
+      });
+
+      const columnMenus = screen.queryAllByRole('button', { name: 'Column menu' });
+      expect(columnMenus.length).toEqual(2);
+      const [_unusedIdColumnMenu, colorColumnMenu] = columnMenus;
+
+      // Act
+      await user.click(colorColumnMenu);
```

Comment:
This is repeated in several tests. Is it worth extracting out into a helper function `openColumnMenu(columnName)` or something like that?

Reply:
Fixed in 4512cf630805eb65e1b9032440fcec78c87d4aa6

I didn't do exactly what you suggested because I would have needed to send `user` down as a param, which I didn't want to do...lmk what you think of the modified solution.